Sources/WordPressKit/Models/Stats/Time Interval/StatsSiteMetricsResponse.swift

```diff
@@ -0,0 +1,108 @@
+import Foundation
+
+public struct StatsSiteMetricsResponse {
+    public var period: StatsPeriodUnit
+    public var periodEndDate: Date
+    public let data: [PeriodData]
+
+    public enum Metric: String, CaseIterable {
+        case views
+        case visitors
+        case likes
+        case comments
+        case posts
+    }
+
+    public struct PeriodData {
+        /// Periods date in the site timezone.
+        public var date: Date
+        public var views: Int?
+        public var visitors: Int?
+        public var likes: Int?
+        public var comments: Int?
+        public var posts: Int?
+
+        public subscript(metric: Metric) -> Int? {
+            switch metric {
+            case .views: views
+            case .visitors: visitors
+            case .likes: likes
+            case .comments: comments
+            case .posts: posts
+            }
+        }
+    }
+}
+
+extension StatsSiteMetricsResponse: StatsTimeIntervalData {
+    public static var pathComponent: String {
+        "stats/visits"
+    }
+
+    public static func queryProperties(with date: Date, period: StatsPeriodUnit, maxCount: Int) -> [String: String] {
+        return [
+            "unit": period.stringValue,
+            "quantity": String(maxCount),
+            "stat_fields": Metric.allCases.map(\.rawValue).joined(separator: ",")
+        ]
+    }
+
+    public init?(date: Date, period: StatsPeriodUnit, jsonDictionary: [String: AnyObject]) {
+        self.init(date: date, period: period, unit: nil, jsonDictionary: jsonDictionary)
+    }
+
+    public init?(date: Date, period: StatsPeriodUnit, unit: StatsPeriodUnit?, jsonDictionary: [String: AnyObject]) {
+        guard let fields = jsonDictionary["fields"] as? [String],
+              let data = jsonDictionary["data"] as? [[Any]] else {
+            return nil
+        }
+
+        guard let periodIndex = fields.firstIndex(of: "period") else {
+            return nil
+        }
+
+        self.period = period
+        self.periodEndDate = date
+
+        let indices = (
+            views: fields.firstIndex(of: Metric.views.rawValue),
+            visitors: fields.firstIndex(of: Metric.visitors.rawValue),
+            likes: fields.firstIndex(of: Metric.likes.rawValue),
+            comments: fields.firstIndex(of: Metric.comments.rawValue),
+            posts: fields.firstIndex(of: Metric.posts.rawValue)
+        )
+
+        let dateFormatter = makeDateFormatter(for: period)
```

Comment:
There is this comment above: "/// Periods date in the site timezone."

It's impossible to parse a date string if we don't know what the site timezone is, right? So, it's incorrect to use any `DateFormatter` instance. I think the correct solution is either passing the site timezone from the app[^1] to the parsing function here, or returning a plain string and letting the app parse it.

[^1]: Or, making another HTTP request to get site timezone, which does feel like wasteful.